pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -151,7 +151,13 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	if err := r.ensureAutomationConfig(mdb); err != nil {
+	authEnabler, err := getAuthenticationEnabler(r.client, mdb)
```

Comment:
to get the authEnabler we need the Kube client and the mdb obect, which is also used everytime we need to pass the authEnabler. Is it posible we make thr `mdb` obect the receiver of `getAuthenticationEnabler` so it can be used and obtained when it is needed and not have to be passed around?

Reply:
You would call `mdb.getAuthenticationEnabler` in `mongodb_controller.go:362` and can avoid passing this object around.

Reply:
yeah I was considering this before, I think it will end up being a lot cleaner, will play around some more with this idea!

Reply:
@rodrigovalin one argument against making the mdb the receiver of this function is that it will no longer purely be responsible for the Kubernetes view of the resource. It will now (inderectly) know how to configure the AutomationConfig. I agree that we should move the getEnabler code into the function that builds the AutomationConfig, but I still think it would be better to have both the client and mdb as arguments to this function. As it is the operator's responsibility to bridge the 2 domains (AC + K8s). Let me know what you think!

Reply:
I agree with that, and I think I don't have a better idea on how to fix the added `enabler` being passed everywhere :(

Reply:
I think we can still create the enabler within the function it is used to minimise its scope. And not need to add to all of the method signatures